packages/blaze/view.js

```diff
@@ -352,13 +352,16 @@ Blaze._materializeView = function (view, parentView, _workStack, _intoArray) {
       view._isInRender = false;
 
       if (! c.firstRun) {
+        // added by ramezrafla@zegenie to skip affecting DOM for unchanged content
+        if (Blaze._deepCompare(lastHtmljs,htmljs)) {
```

Comment:
Oh, I just used deepCompare for the sake of convenience and because I haven't gone through all the options. deepCompare short circuits comparison if it's a primitive so I am reusing the same function for risk mitigation as the data level.

But I did see somewhere in the code that htmljs can be views. Also, we skipped DOM comparison altogether that was there before, which was my biggest concern. If I was too cautious, then we can simply simplify the compare function. But, why would the original developers convert to DOM if it was always primitives?